homeassistant/components/energy/sensor.py

```diff
@@ -205,7 +206,8 @@ def _update_cost(self) -> None:
         else:
             # Update with newly incurred cost
             old_energy_value = float(self._last_energy_sensor_state.state)
-            self._attr_state = cur_value + (energy - old_energy_value) * energy_price
+            self._cur_value += (energy - old_energy_value) * energy_price
```

Comment:
There is another functional change in here, which I'm not sure if it was done on purpose or not.

The value is now only calculated for the changed part and next summed on the existing value. Previously, it would calculate it over whole value.

The difference is, is that if the price is correct, it will only be applied to the new differences, not the total value.

Reply:
Not intended to change anything else than rounding
Do not think I understand what you mean here.

Reply:
NVM it was the order that throws me off in the old version. It does absolutely the same. 👍